CDP4Dal/Assembler.cs

```diff
@@ -204,25 +214,39 @@ public async Task Synchronize(IEnumerable<CDP4Common.DTO.Thing> dtoThings, bool
 
                     foreach (var dtoThing in this.DtoThingToUpdate)
                     {
-                        Lazy<Thing> updatedLazyThing;
                         var cacheKey = new CacheKey(dtoThing.Iid, dtoThing.IterationContainerId);
-                        var succeed = this.Cache.TryGetValue(cacheKey, out updatedLazyThing);
+                        var succeed = this.Cache.TryGetValue(cacheKey, out var updatedLazyThing);
                         
                         if (succeed)
                         {
                             var thingObject = updatedLazyThing.Value;
                             var cacheId = new CacheKey(dtoThing.Iid, dtoThing.IterationContainerId);
+
                             if (!existentGuid.Select(x => x.Item1).Contains(cacheId))
                             {
                                 CDPMessageBus.Current.SendObjectChangeEvent(thingObject, EventKind.Added);
                                 messageCounter++;
                             }
-                            else if (dtoThing.RevisionNumber > existentGuid.Single(x => x.Item1.Equals(cacheId)).Item2)
+                            else
                             {
-                                // send event only if revision number has increased from the old cached version
-                                CDPMessageBus.Current.SendObjectChangeEvent(thingObject, EventKind.Updated);
-                                messageCounter++;
-                            }               
+                                var cacheThingRevisionNumber = existentGuid.Single(x => x.Item1.Equals(cacheId)).Item2;
+
+                                if (dtoThing.RevisionNumber > cacheThingRevisionNumber)
+                                {
+                                    // send event only if revision number has increased from the old cached version

```

Comment:
the comment on line 236 and line 244 are a bit confusing because it states that the event is `only`  sent when a condition is met.. .but it seems not to be `only`  for the described event. Please clarify or fix

Reply:
Old comment, that is indeed not valid anymore. 
I'll remove the word ```only```.